docs/design/specs/Ecma-335-Augments.md

```diff
@@ -948,3 +949,47 @@ Conversions from floating-point numbers to integral values truncate the number t
 on the top of the stack is reinterpreted as an unsigned value before the conversion.
 Note that integer values of less than 4 bytes are extended to int32 (not native int) on the
 evaluation stack.
+
+## Ref Fields
+To improve the usefulness of ref structs, support for fields which are defined as byrefs is needed. Currently their functionality can be approximated by Span<T> fields, but not all types can be converted into Span<T> types simply. In order to support these customers support for generalizing byref fields, and converting TypedReference, ArgIterator and RuntimeArgumentHandle into following the normal rules of C# ref structs is desired. With this set of changes, it becomes possible to have ByRef fields of T, but support for pointers to ByRef fields or ByRefs to ByRefs is not added to the ECMA specification.
```

Comment:
What does this mean in the context of things like `Span<T>*` and `ref Span<T>`, both of which are legal today?

Does it just mean you could only declare it "indirectly" (e.g. `ref SomeStructWithRefField`) or is there some other "subtlety" here?

Reply:
Yes, you could only declare it indirectly. There are a bunch of weird and unfortunate costs to adding the ability to take a ref of a ref, or a pointer to a ref.
- Mono would need to restructure its representation of type
- We would need to update the type name generation/parsing for custom attribute types
- A bunch of little implementation details scattered all throughout the CoreCLR runtime would need updates.
- We would need new syntax in C#, or there may be a whole set of unspeakable types that can only appears as a result of generics or something

Given that, I'd like to make the first effort to implement this simply not allow that more complex feature to exist. I don't currently see a need for it in the motivation for this feature. Also, given that is just a strict restriction on behavior, if it turns out to be a capability we want to add in the future, we can do so.